src/Jackalope/Repository.php

```diff
@@ -155,6 +157,10 @@ public function login(CredentialsInterface $credentials = null, $workspaceName =
             $session->getWorkspace()->setTransactionManager($utx);
         }
 
+        if ($this->transport instanceof GenericVersioningInterface) {
+            $this->transport->setVersionHandler(new VersionHandler($session));
```

Comment:
i guess this is more of a jackalope 2.0 topic, but we go up and down the layers here.

maybe an input for @dantleech to think about how we could improve the general architecture. (also, it looks like we could use a simple DI container like pimple in 2.0 to make such things more elegant)